src/gordon_gcp/clients/gdns.py

```diff
@@ -110,15 +110,41 @@ def _parse_resp_to_records(self, response, records):
             rrset = GCPResourceRecordSet(**record)
             records.append(rrset)
 
+    def get_managed_zone(self, zone):
+        """Get the GDNS managed zone name for a DNS zone.
+
+        Google uses custom string names with specific `requirements
+        <https://cloud.google.com/dns/api/v1/managedZones#resource>`_
+        for storing records. The scheme implemented here chooses a
+        managed zone name which removes the trailing dot and replaces
+        other dots with dashes, and in the case of reverse records,
+        uses only the two most significant octets, prepended with
+        'reverse'.
+
+        Example:
+           get_managed_zone('example.com.') = 'example-com'
+           get_managed_zone('30.20.10.in-addr.arpa.) = 'reverse-20-10'
+
+        Args:
+            zone (str): DNS zone.
+        Returns:
+            str of managed zone name.
+
+        """
+        if zone.endswith('.in-addr.arpa.'):
+            return 'reverse-' + '-'.join(zone.split('.')[1:3])
+        return '-'.join(zone.split('.')[:-1])
+
     async def get_records_for_zone(self, zone):
         """Get all resource record sets for a particular managed zone.
```

Comment:
This should probably be 'managed zone'.

Reply:
Do you mean "*not* be managed zone"?  I was on the fence, but I can remove it, if you like.

Reply:
i think @ocsw was referring to the function name?